nsxt/resource_nsxt_transport_node.go

```diff
@@ -0,0 +1,1959 @@
+			Description: "the Node maintained on the Node and used to recognize the Node",
+		},
+		"fqdn": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Fully qualified domain name of the fabric node",
+		},
+		"id": getNsxIDSchema(),
+		"ip_addresses": {
+			Type:     schema.TypeList,
+			Optional: true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+	}
+
+	maps.Copy(elemSchema, addlAttributes)
+	s := schema.Schema{
+		Type:     schema.TypeList,
+		MaxItems: 1,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: elemSchema,
+		},
+	}
+	if addExactlyOneOf {
+		s.ExactlyOneOf = []string{
+			"edge_node",
+			"host_node",
+			"public_cloud_gateway_node",
+		}
+	}
+	return &s
+}
+
+func getEdgeNodeDeploymentConfigSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		MaxItems:    1,
+		Optional:    true,
+		Description: "Config for automatic deployment of edge node virtual machine",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"form_factor": {
+					Type:         schema.TypeString,
+					Default:      model.EdgeNodeDeploymentConfig_FORM_FACTOR_MEDIUM,
+					Optional:     true,
+					ValidateFunc: validation.StringInSlice(edgeNodeFormFactorValues, false),
+				},
+				"node_user_settings": {
+					Type:     schema.TypeList,
+					MaxItems: 1,
+					Required: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"audit_password": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Sensitive:   true,
+								Description: "Node audit user password",
+							},
+							"audit_username": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Description: "CLI \"audit\" username",
+							},
+							"cli_password": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Sensitive:   true,
+								Description: "Node cli password",
+							},
+							"cli_username": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Default:     "admin",
+								Description: "CLI \"admin\" username",
+							},
+							"root_password": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Sensitive:   true,
+								Description: "Node root user password",
+							},
+						},
+					},
+				},
+				"vm_deployment_config": {
+					Type:     schema.TypeList,
+					MaxItems: 1,
+					Required: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"compute_folder_id": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Description: "Compute folder identifier in the specified vcenter server",
+							},
+							"compute_id": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "Cluster identifier or resourcepool identifier for specified vcenter server",
+							},
+							"data_network_ids": {
```

Comment:
If this can be segment path, perhaps we should call it `data_networks` to avoid confusion

Reply:
AFAIK this is an id of a vSphere network object